File: frontend/src/app/shared/model/api/OkrUnit/okr-topic-draft.dto.ts
import { OkrUnitId } from '../../id-types';
import { OkrTopicDescriptionDto } from './okr-topic-description.dto';
import { User } from '../user';

export class OkrTopicDraftDto extends OkrTopicDescriptionDto {
  okrParentUnitId: OkrUnitId;
  currentStatus: number;
  initiator: User;
}

File: backend/burning-okr/burning-okr-app/src/main/java/org/burningokr/dto/okr/OkrTopicDraftDto.java
package org.burningokr.dto.okr;

import jakarta.validation.constraints.PositiveOrZero;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.burningokr.model.users.User;

@Data
@EqualsAndHashCode(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
public class OkrTopicDraftDto extends OkrTopicDescriptionDto {
  private Long okrParentUnitId;
  private User initiator;

  @PositiveOrZero
  private int currentStatus;


  public OkrTopicDraftDto(OkrTopicDescriptionDto parentUnit) {
    super(parentUnit);
  }
}
